src/zope/traversing/api.py

```diff
@@ -223,3 +182,9 @@ def canonicalPath(path_or_object):
 
 # import this down here to avoid circular imports
 from zope.traversing.adapters import traversePathElement
+
+# Synchronize the documentation.
+for name in ITraversalAPI.names():
+    if name in globals():
+        globals()[name].__doc__ = ITraversalAPI[name].__doc__
+del name
```

Comment:
Oh, I see!

I like the reduction in duplication, but maybe add "See ITraversalAPI for more details" to each of the docstrings that are replaced in this fashion?  Some people (e.g. me) will looking at the source code rather than read the docs on a web page, and the loss of detail should be offset by a hint where to find more detail.

Reply:
Good idea! I agree it's important to find a balance.